scripts/generate-app-config/generateRouteConfig.js

```diff
@@ -0,0 +1,205 @@
+    let menu = acc.menu;
+    const menuItems = acc.menuItems || [];
+    const hasSubMenu = page.pages && page.pages.length > 0;
+
+    const routePath = `${rootPath}${page.path}`;
+    let redirectRoute;
+
+    // if the given page, has sub menu items, add them to the overall route object.
+    if (hasSubMenu) {
+      // recursively call to get child content, and menu items
+      const { content: childContent, menu: childMenu, menuItems: childMenuItems } = generateRouteConfig(page.pages, routePath, placeholder, routeImporter);
+
+      content = Object.assign(content, childContent);
+      menu = Object.assign(menu, childMenu);
+
+      // Add a menu item containing links to the child content.
+      menu[routePath] = routeItem(routePath, { contentPath: RoutingMenu, name: 'RoutingMenu' }, menuProps(page.name, childMenuItems), routeImporter);
+
+      // If the page does not have content, but the first submenu item has content, but not a menu. Redirect directly to that item.
+      const subPage = page.pages[0];
+      if (subPage.content && !subPage.pages && !page.content) {
+        redirectRoute = `${rootPath}${page.path}${subPage.path}`;
+      }
+    }
+
+    // provide the menu item for this content page.
+    menuItems.push(menuItem(page.name, routePath, hasSubMenu));
+
+    // If the pages has content, add the content render item. If not, add a placeholder item.
+    if (page.content) {
+      content[routePath] = contentRouteItem(routePath, { contentPath: page.content }, page.props, page.type, routeImporter);
+    } else if (redirectRoute) {
+      // If a redirect Route has been identified, redirect to it.
+      content[routePath] = contentRouteItem(routePath, { contentPath: Redirect, name: '{ Redirect }', identifier: 'Redirect' }, { to: redirectRoute }, 'js', routeImporter);
+    } else {
+      content[routePath] = contentRouteItem(routePath, { contentPath: placeholder.content, name: 'TerraDevSitePlaceholder' }, placeholder.props, 'js', routeImporter);
+    }
+
+    return { content, menu, menuItems };
+  }, { content: {}, menu: {} })
+);
+
+/**
+* Create a page config object for the top level page.
+*/
+const getPageConfig = (name, pagePath, pages, type, siteConfig, routeImporter) => {
+  const { readMeContent } = siteConfig;
+  const config = {
+    name: startCase(name),
+    path: pagePath,
+    pages,
+  };
+
+  // Special logic to add a home component with a readme if readme content is provided in site config and no other home items are found.
+  if (type === 'home' && readMeContent) {
+    config.content = TerraDocTemplate;
+    config.props = { readme: routeImporter.addImport(ImportAggregator.relativePath(readMeContent)) };
+  }
+
+  return config;
+};
+
+/**
+* Build out the page config for the top level link.
+*/
+const getLinkRoute = (link, pageConfig, siteConfig, routeImporter) => {
+  let pages = [];
+  let type;
+
+  // If a link has more than one type, we'll have to build out at least three page configs.
```

Comment:
> we'll have to build out at least three page configs.

this is weird, could you define the types? I'm assuming each type plus raw?

Reply:
navigation items can now support an array of page types, which will be added to the to of the page config as menu items. So by default we will have one page config generated, but if there is more than one page type we will build out one config for the nav type and one for each page type.